lib/facter/resolvers/windows/processors.rb

```diff
@@ -19,14 +19,20 @@ def post_resolve(fact_name, _options)
 
         def read_fact_from_win32_processor(fact_name)
           win = Facter::Util::Windows::Win32Ole.new
-          proc = win.exec_query('SELECT Name,Architecture,NumberOfLogicalProcessors FROM Win32_Processor')
+          query_string = 'SELECT Name,'\
+          'Architecture,'\
+          'NumberOfLogicalProcessors,'\
+          'ThreadCount,'\
+          'NumberOfCores FROM Win32_Processor'
+          proc = win.exec_query(query_string)
```

Comment:
I think this query does not work anymore, I'm getting this on Windows 2012R2:

```
C:\ProgramData\PuppetLabs>facter processors
{
  cores => 0,
  count => 0,
  models => [

  ],
  physicalcount => 0,
  threads => 0
}
```

Jenkins seems to fail for the same reason

Reply:
The issue was that `ThreadCount` wasn't a valid wmic cpu query for windows2012r2, so I removed it and used `NumberOfLogicalProcessors` as the total number of threads instead, which also solves a previous issue where ThreadCount queries would return 0 or nil in windows virtual machines.